src/compiler/checker.ts

```diff
@@ -15473,13 +15603,27 @@ namespace ts {
                 bindSourceFile(file, compilerOptions);
             });
 
+            let mergeAugmentations = false;
             // Initialize global symbol table
             forEach(host.getSourceFiles(), file => {
                 if (!isExternalOrCommonJsModule(file)) {
                     mergeSymbolTable(globals, file.locals);
                 }
+                mergeAugmentations = mergeAugmentations || file.moduleAugmentations.length > 0;
             });
 
+            if (mergeAugmentations) {
+                // merge module augmentations.
+                // this needs to be done after global symbol table is initialized to make sure that all ambient modules are indexed 
+                for (const file of host.getSourceFiles()) {
+                    if (file.moduleAugmentations.length) {
+                        for (const augmentation of file.moduleAugmentations) {
+                            mergeModuleAugmentation(augmentation);

```

Comment:
Well you have a function that can act as a simple callback, so it seems unnecessary to explicitly write the loop